src/Components/ExternalResult/ResultList.tsx

```diff
@@ -158,8 +159,8 @@ export default function ResultList() {
   };
 
   let resultList: any[] = [];
-  if (data && data.length) {
-    resultList = data.map((result: any) => {
+  if (data && resultListData.length) {
+    resultList = resultListData.map((result: any) => {
```

Comment:
consider using `PaginatedList` for listing (see how `BedManagement.tsx` is implemented.